js/rollup.config.ts

```diff
@@ -0,0 +1,98 @@
+import alias from '@rollup/plugin-alias';
+import commonJsPlugin from '@rollup/plugin-commonjs';
+import nodeResolve from '@rollup/plugin-node-resolve';
+import replace from '@rollup/plugin-replace';
+import fs from 'fs';
+import * as path from 'path';
+import type { RollupOptions } from 'rollup';
+import externals from 'rollup-plugin-node-externals';
+import ts from 'rollup-plugin-ts';
+
+function createConfig({
+    bundleName,
+    format,
+    runtime,
+}: {
+    bundleName: string;
+    format: 'cjs' | 'esm';
+    runtime: 'browser' | 'node' | 'react-native';
+}): RollupOptions {
+    return {
+        input: 'src/index.ts',
+        output: {
+            file: 'lib/' + format + '/' + bundleName,
+            format,
+        },
+        plugins: [
+            alias({
+                entries: [
+                    {
+                        find: /^\./, // Relative paths.
+                        replacement: '.',
+                        async customResolver(source, importer, options) {
+                            const resolved = await this.resolve(source, importer, {
+                                skipSelf: true,
+                                ...options,
+                            });
+                            if (resolved == null) {
+                                return;
+                            }
+                            const { id: resolvedId } = resolved;
+                            const directory = path.dirname(resolvedId);
+                            const moduleFilename = path.basename(resolvedId);
+                            const forkPath = path.join(directory, '__forks__', runtime, moduleFilename);
```

Comment:
None of this is relevant. You won't have forks; this library will only ever be for React Native.

Reply:
tbh, I have a very limited understanding of how any of this rollup/ts config stuff works. I copied fiels from the MWA repo until things started building successfully lol. I'll backtrack through and clean all this up, but might do that in a separate PR.

Reply:
@Michaelsulistio should I refactor this lib in the same way you did the walletlib [here](https://github.com/solana-mobile/mobile-wallet-adapter/pull/479)?

Reply:
Yup you can pretty much copy what I did there, although I used commonJS as that was the default in the bob builder config I copied over.